packages/cli/src/environments.ee/source-control/source-control-import.service.ee.ts

```diff
@@ -523,6 +531,95 @@ export class SourceControlImportService {
 		return { tags: localTags, mappings: localMappings };
 	}
 
+	/**
+	 * Reads projects from the git work folder and returns the projects that are accessible to the context user
+	 */
+	async getRemoteProjectsFromFiles(
+		context: SourceControlContext,
+	): Promise<ExportableProjectWithFileName[]> {
+		const remoteProjectFiles = await glob('*.json', {
+			cwd: this.projectExportFolder,
+			absolute: true,
+		});
+
+		const remoteProjects = await Promise.all(
+			remoteProjectFiles.map(async (file) => {
+				this.logger.debug(`Parsing project file ${file}`);
+				const fileContent = await fsReadFile(file, { encoding: 'utf8' });
+				const parsedProject = jsonParse<ExportableProject>(fileContent);
+
+				return {
+					...parsedProject,
+					filename: getProjectExportPath(parsedProject.id, this.projectExportFolder),
+				};
+			}),
+		);
+
+		if (context.hasAccessToAllProjects()) {
+			return remoteProjects;
+		}
+
+		const accessibleProjects =
+			await this.sourceControlScopedService.getAuthorizedProjectsFromContext(context);
+
+		return remoteProjects.filter((remoteProject) => {
+			return findOwnerProject(remoteProject.owner, accessibleProjects);
+		});
+	}
+
+	/**
+	 * Fetches team projects from the database that are accessible to the context user
+	 */
+	async getLocalTeamProjectsFromDb(
+		context: SourceControlContext,
+	): Promise<ExportableProjectWithFileName[]> {
+		const localProjects = await this.projectRepository.find({
+			select: ['id', 'name', 'description', 'icon', 'type'],
+			where: {
+				type: 'team',
+				...(this.sourceControlScopedService.getProjectsWithPushScopeByContextFilter(context) ?? {}),
+			},
+		});
+
+		return localProjects.map((local) =>
+			this.mapProjectEntityToExportableProjectWithFileName(local),
+		);
+	}
+
+	/**
+	 * Fetches all team projects from the database, regardless of the context user's access
+	 */
+	async getAllTeamLocalProjectsFromDb(): Promise<ExportableProjectWithFileName[]> {
```

Comment:
I wonder if we could merge with the previous one, making the context optional or something